src/terminal.ts

```diff
@@ -23,26 +28,30 @@ export default class Terminal {
 		this.command = commandPath;
 	}
 
-	public runCommand (args: string[]) {
-		if (!this.terminal) {
-			this.terminal = window.createTerminal({ name: this.name });
-		}
+	public runCommand (args: string[], { forceTerminal = false } = {} ) {
+		if (ExtensionContainer.config.general.useTerminalForBuild || forceTerminal) {
+			if (!this.terminal) {
+				this.terminal = window.createTerminal({ name: this.name });
+			}
 
-		const activeTerminal = window.activeTerminal;
-		// Only call show if we arent the active terminal
-		if (!activeTerminal || activeTerminal.name !== this.terminal.name) {
-			this.terminal.show();
+			const activeTerminal = window.activeTerminal;
+			// Only call show if we arent the active terminal
+			if (!activeTerminal || activeTerminal.name !== this.terminal.name) {
+				this.terminal.show();
+			}
+			this.clear();
+			this.terminal.sendText(`${this.command} ${args.join(' ')}`);
+		} else {
+			this.runCommandInOutput(args);
 		}
-		this.clear();
-		this.terminal.sendText(`${this.command} ${args.join(' ')}`);
 	}
 
 	public runCommandInBackground (args: string[], progressOptions: ProgressOptions = { location: ProgressLocation.Window }, spawnOptions: SpawnOptions = { shell: true }) {
 		return window.withProgress(progressOptions, () => {
 			return new Promise((resolve, reject) => {
-				const proc = spawn(this.command, args, spawnOptions);
+				this.proc = spawn(this.command, args, spawnOptions);
```

Comment:
I think it'd be best to maybe not assign the process created here to `this.proc`.

This function is currently used by the create app/module commands and I think it's fine to have those be usable when a build is running. Long term we might move things like alloy component generation to use this too, which we would want to be usable during a build.